bool detect_obj
---
string obj
